airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -95,6 +98,14 @@ def raise_on_http_errors(self) -> bool:
         # never raise on http_errors because this overrides the error handler logic...
         return False
 
+    @property
+    def max_retries(self) -> Union[int, None]:
+        if self.disable_retries:
+            return 0
+        if hasattr(self.requester.error_handler, "max_retries"):
```

Comment:
CustomErrorHandler might not expose the value so we need to validate if the attribute exists

Reply:
How would you suggest to do that? I don't see any way of enforcing the usage of an interface when implementing a CustomErrorHandler

Reply:
Fine to keep out of scope of this PR, but we should aim to be in a place where we can assume custom components implement the interface they're supposed to - custom error handlers should implement the error handler interface, and custom paginators should implement the paginator interface.

One approach would be for the [`create_custom_component`](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py#L269) to verify that the class inherits the expected interface.

Reply:
[Created an issue](https://github.com/airbytehq/airbyte/issues/25463). No need to worry about it here